src/asmcup/runtime/World.java

```diff
@@ -8,9 +8,11 @@
 	protected final HashMap<Integer, byte[]> tileData;
 	protected final int seed;
 	protected int frame;
+
+	private static final Random random = new Random();
 	
 	public World() {
-		this(new Random().nextInt());
+		this(random.nextInt());

```

Comment:
Math.random has a crooked distribution for ints. Shouldn't matter that much, but doesn't hurt either